From 59b0b396f68437a655dbd382907a4d22f8458d51 Mon Sep 17 00:00:00 2001
From: Eric Naim <dnaim@cachyos.org>
Date: Mon, 18 May 2026 13:24:30 +0800
Subject: [PATCH 18/33] platform/x86: hp-wmi: Add support for OMEN Slim
 16t-an000 (8D40)

Similar to other Victus S devices, this device requires the same WMI
queries for platform profiles but with disabled EC readback as on board
8D41 and 8D87.

Add this device to the list so platform profiles can be registered successfully.

Signed-off-by: Eric Naim <dnaim@cachyos.org>
---
 drivers/platform/x86/hp/hp-wmi.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/platform/x86/hp/hp-wmi.c b/drivers/platform/x86/hp/hp-wmi.c
index 37937d8a0..f3fc95f88 100644
--- a/drivers/platform/x86/hp/hp-wmi.c
+++ b/drivers/platform/x86/hp/hp-wmi.c
@@ -253,6 +253,10 @@ static const struct dmi_system_id victus_s_thermal_profile_boards[] __initconst
 		.matches = { DMI_MATCH(DMI_BOARD_NAME, "8D26") },
 		.driver_data = (void *)&omen_v1_legacy_thermal_params,
 	},
+	{
+		.matches = { DMI_MATCH(DMI_BOARD_NAME, "8D40") },
+		.driver_data = (void *)&omen_v1_no_ec_thermal_params,
+	},
 	{
 		.matches = { DMI_MATCH(DMI_BOARD_NAME, "8D41") },
 		.driver_data = (void *)&omen_v1_no_ec_thermal_params,
-- 
2.55.0

